### PR TITLE
storage: improve conflicting split error messsage

### LINKED ISSUE #5629: storage: the message for range split conflicts should be improved
Reported by @yznming on Gitter:

```
E160325 20:14:13.922965 storage/queue.go:444  [split] range=23 [/Table/51/1/2360159016-/Table/51/1/2725786584): error: storage/replica_command.go:1664: split at key /Table/51/1/2725788664 failed: unexpected value: raw_bytes:"(\016J\035\003\010\027\022\007\273\211\371\214\255+(\032\007\273\211\371\242x3\330\"\006\010\001\020\001\030\001\"\006\010\004\020\004\030\002\"\006\010\007\020\007\030\003(\004" timestamp:<wall_time:1458908049375339590 logical:1 >
```

As suggested below by Tobias this message indicates a conflict during a range split. However the text of the message does not convey this information clearly. The message should be improved.

### PR DESCRIPTION
**Pull request by petermattis**
Fixes #5629

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8047)

### CODE REVIEW

File: storage/client_split_test.go [hunk 1]
```diff
@@ -279,21 +280,31 @@ func TestStoreRangeSplitConcurrent(t *testing.T) {
...
+
+	var failureCount int
+	for i := 0; i < concurrentCount; i++ {
+		pErr := <-errCh
+		if pErr != nil {
+			// There are only two expected errors from concurrent splits: conflicting
+			// range descriptors if the splits are initiated concurrently or that the
+			// range is already split at the specified key.
+			s := pErr.String()
+			if strings.Index(s, "conflict updating range descriptors") == -1 &&
```
Comment (tamird) [inline storage/client_split_test.go]: GoError + testutils?

Comment (petermattis) [inline storage/client_split_test.go]: Sure. Done.

**Comment by tamird** [comment]
lgtm

File: storage/replica_command.go [hunk 2]
```diff
@@ -2336,6 +2336,9 @@ func (r *Replica) AdminSplit(
 			return err
 		}
 		if err := txn.Run(b); err != nil {
+			if _, ok := err.(*roachpb.ConditionFailedError); ok {
+				return errors.Errorf("conflict updating range descriptors")
```
Comment (tschottdorf) [inline storage/replica_command.go]: no `errors.Wrap`?

**Comment by tschottdorf** [comment]
LGTM

File: storage/replica_command.go (hunk 2 above)
Comment (petermattis) [inline storage/replica_command.go]: Yeah, but do we need to include the original error. Using `errors.Wrapf` the error prints as:

```
expected error: storage/replica_command.go:2368: split at key "a" failed: conflict updating range descriptors: unexpected value: raw_bytes:"I\205B\247\003\010\001\022\000\032\001a\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:123 logical:70 >
```

vs `errors.Errorf`:

```
expected error: storage/replica_command.go:2368: split at key "a" failed: conflict updating range descriptors
```

Comment (tschottdorf) [inline storage/replica_command.go]: Assuming that that message contained some more context, I'd have a hard time figuring out when I would actually need the extra bit. I had #7600 and comrades in the back of my mind. Should be fine to omit this, though.

**Pull request merged**